python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -300,6 +301,29 @@ def project_name(self) -> str:
             raise DgError("`project_name` is only available in a Dagster project context")
         return self.root_path.name
 
+    def get_module_version(self, module_name: str) -> str:
+        if not self.use_dg_managed_environment:
+            raise DgError("`get_module_version` is only available in a Dagster project context")
+
+        with pushd(self.root_path):
+            result = subprocess.check_output(
+                [
+                    "uv",
```

Comment:
I think we need to stop assuming `uv` available everywhere all the time. @smackesey do we have an abstraction layer of some sort to enforce this beyond tests?

Reply:
Sort of-- python executables (like `dagster-dg`) should be resolved through `DgContext.get_executable`. This is a little bit different since we are either going to use `uv` (globally installed) or `pip` (part of a Python environment). So I think for this special case we need to add another method like `DgContext.resolve_package_manager_executable()` or something that returns either `["uv", "pip"]` or does `DgContext.get_executable` if `uv` isn't available.

Reply:
Ok let's add `resolve_package_manager_executable`